lib/remote/apilistener-configsync.cpp

```diff
@@ -104,6 +107,15 @@ Value ApiListener::ConfigUpdateObjectAPIHandler(const MessageOrigin::Ptr& origin
 		return Empty;
 	}
 
+	// Wait for the object name to become available for processing.
+	m_ConfigSyncPolicy->WaitForObject(ptype, objName);
+	// Add object name to the locked list to block all other threads that try to process a message affecting the same object.
+	m_ConfigSyncPolicy->BlockObject(ptype, objName);
```

Comment:
Those two calls should be a single operation. As both calls lock individually, other threads might interfere in between, i.e. a race condition.